### ISSUE TITLE
CSS tweak: scroll to top of toc visually as well as in DOM

### ISSUE THREAD

**Issue by SallyMcGrath** [CodeYourFuture/curriculum#57]
Looks good, but currently appears to scroll to the middle of the TOC rather than the top? Any idea why?

This is what I see after scrolling down and clicking the link:

<img width="1677" alt="Screenshot 2023-06-11 at 11 01 08" src="https://github.com/CodeYourFuture/curriculum/assets/1131704/eaf3060b-a204-478c-b422-8b9789c2625c">

_Originally posted by @illicitonion in https://github.com/CodeYourFuture/curriculum/issues/50#issuecomment-1586097981_

**Comment by 40thieves** [comment]
This is because it's scrolling the main content to align with the top of the TOC. The nav bar then covers over the top. If you artificially hide the header, then you can see that it does align correctly:

<img width="1217" alt="Screenshot 2023-06-11 at 16 06 22" src="https://github.com/CodeYourFuture/curriculum/assets/424411/04998fe7-f003-4418-a085-7b52d7ad793d">

(Ignore the weird fonts - I'm on train wifi and it doesn't seem to like Google Fonts)

I'm 90% sure there's a simple CSS solution, I thought it was either [`scroll-padding`](https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-padding) or [`scroll-margin`](https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-margin), but in some testing I don't seem to be able to get it to work as I thought it did. It's possible it's another property that I'm failing to find right now. Or possible that train wifi is interfering here too (it does seem to cause issues with `#`-scrolling).

**Comment by 40thieves** [comment]
Huh, this is weird. It seems like `scroll-margin` does work in Firefox & Safari, but not Chrom(ium) - but only in the syllabus. But weirdly [this demo](https://codepen.io/jh3y/pen/QWKRzbd) I found seems to work in Chrome too. I guess it's something to do with how our layout works?

I guess I can send a PR that only works in FF/Safari, and then try to come up with a minimal example of where it breaks?

**Comment by 40thieves** [comment]
Ok, I've pushed [a branch](https://github.com/CodeYourFuture/curriculum/compare/bugfix/top-to-toc...bugfix/toc-scroll-margin) with the `scroll-margin` change as mentioned above. Unfortunately it's a bit ugly as we don't have a fixed header height - it flows from the logo, which is has a fixed width but not height. And, as mentioned above, it only seems to work in Firefox & Safari.

@SallyMcGrath I'll leave it up to you whether you want to pull that commit into this branch 🙂

**Comment by SallyMcGrath** [comment]
Oh yes I did know why ;)  (it's Daniel's text from the linked comment not mine)

I've updated your branch with a clearer fix. The issue is the perennial overflow interaction that so often messes up sticky as well. To fix it I wrapped the toc with the anchor and put the overflow inside it. 

I also posted the header height to make it available to root/everywhere. It flows from the menu button I think, which is the touch target sizing. 

You've branched from a version with the old header in, though, so suggest updating before checking. I just added double the header height but you might want to do something different.
